src/table.js

```diff
@@ -538,15 +543,99 @@ export function getTypeValidator(colType) {
   }
 }
 
+export function coerceToType(value, type, options = {}) {
+  const defaultValue = options.soft ? value : null;
+  const numberDefault = defaultValue === null ? NaN : defaultValue;
+  switch (type) {
+    case "string":
+      return value === "string"
+        ? value.trim()
+        : value || value === 0
+        ? value.toString()
+        : defaultValue;
+    case "boolean":
+      return value === true || value === "true"
+        ? true
+        : value === false || value === "false"
+        ? false
+        : defaultValue;
+    case "integer":
+      return !value || isNaN(parseInt(value)) ? numberDefault : parseInt(value);
+    case "bigint":
+      return typeof value === "bigint"
+        ? value
+        : !value || isNaN(value)
+        ? numberDefault
+        // eslint-disable-next-line no-undef
+        : BigInt(value);
+    case "number": {
+      return !value || isNaN(value) ? numberDefault : Number(value);
+    }
+    case "date": {
+      if (value instanceof Date) return value;
+      if (typeof value === "string") {
+        let match;
+        if (
+          (match = value.match(
+            /^([-+]\d{2})?\d{4}(-\d{2}(-\d{2})?)?(T\d{2}:\d{2}(:\d{2}(\.\d{3})?)?(Z|[-+]\d{2}:\d{2})?)?$/
+          ))
+        ) {
+          if (fixTz && !!match[4] && !match[7])
+            value = value.replace(/-/g, "/").replace(/T/, " ");
+        }
+      }
+      // Invalid Date objects are still instances of Date, but they return true
+      // from isNaN(). If we are "soft-coercing," we want to return the original
+      // value for invalid dates. Otherwise, if a date is invalid, return an
+      // Invalid Date object.
+      const date = new Date(value);
+      const dateDefault = options.soft ? value : date;
+      return isNaN(date) ? dateDefault : date;
+    }
+    case "array":
+      if (Array.isArray(value)) return value;
+      return value && Array.isArray(Array.from(value))
+        ? Array.from(value)
+        : defaultValue;
+    case "buffer":
+    case "object":
+    case "other":
```

Comment:
Any suggestions for how else we could handle coercing to array, buffer, or other?

Reply:
I'm sure you've considered this, but would it be expected to take the value and put it into an array or object?
```js
case "array":
   return [defaultValue]
case "object"
  return {value: defaultValue}
```

I imagine this could be useful when using the data table cell for data transformations (i.e., so that downstream values all have the specified type). Not yet super helpful for representing values in the table, as we don't have great displays for these types.

Reply:
We would want `value` rather than `defaultValue` (as `defaultValue` is only supposed to be returned in the event of coercion failure), but otherwise, sure, perhaps that's a fine way to handle them. Will update.